File: 53-maximum-subarray/maximum-subarray.js

/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let maxSub = nums[0];
    let curSum = 0;

    for (let i = 0; i < nums.length; i++) {
        curSum = Math.max(nums[i], curSum + nums[i]);
        maxSub = Math.max(maxSub, curSum);
    }
    return maxSub;
};

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSub = nums[0];
        int curSum = 0;

        for (int num : nums) {
            curSum = max(num, curSum + num);
            maxSub = max(maxSub, curSum);
        }
        return maxSub;
    }
};